### ISSUE TITLE
Bug: [no-misused-spread] report for string spread harmfully conflates Unicode code points with UTF-16 code units

### ISSUE THREAD

**Issue by andersk** [typescript-eslint/typescript-eslint#10745]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.6.3&fileType=.tsx&code=MYewdgzgLgBBMF4YCJBd4Ycw3Bvu4Gx1kG4AoA0SEAGwFMA6MkAcwAoApAZQHkA5K6AJwEswdPgDMAngwDaVaRAC6ASnmEA9MpgBhEDx4VgUAFwwJaZABpkWM8lyyCqmACEArrDAgAtAFs%2BEJxAoAJu4QAA46AIYBMKCeIWThAvBQABYUOqYw4WBRvnR0FNAQ%2BkQkEOTUtIysnNxQ-IIi4hDccXxQDMjIikp2agCSYKDaugZGJuYAOk4BABwAbADsVlMBAcJzVjYEQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oFtLlZkiACa1kxaIgCGQ9FETRoHaJHBgAviDVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const s = "螺𩷶屬";

console.log(JSON.stringify([...s]));
// Correct: ["螺","𩷶","屬"]
// But no-misused-spread complains there, and suggests:

console.log(JSON.stringify(s.split("")));
// Incorrect: ["螺","\ud867","\uddf6","屬"]
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-misused-spread": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

In JavaScript, string spreading is the correct way to split a string into Unicode code points. It should not be an error.

### Actual Result

``error  Using the spread operator on a string can cause unexpected behavior. Prefer `.split('')` instead.  @typescript-eslint-no-misused-spread``

But `.split('')` splits a string into UTF-16 code units, not Unicode code points. For characters outside the basic multilingual plane, this produces invalid unpaired surrogates.

### Additional Info

This rule was introduced with a justification linking to the blog post “[Strings Shouldn’t Be Iterable By Default](https://www.xanthir.com/b4wJ1)”. But this post is about how to design better languages in the future, not how to use the languages we already have.

In the JavaScript language we have, `[...s]` *is the correct way* to split a string into Unicode code points, and `.split('')` does not do that correctly. The suggestion is harmful and will result in the introduction of subtle internationalization bugs triggered with less common languages (and emoji).

There isn’t another good way to correctly split a string into Unicode code points, so this rule should just be disabled for strings.

**Comment by Josh-Cena** [comment]
Wait, I don't remember this being our consensus when we introduced it in https://github.com/typescript-eslint/typescript-eslint/pull/10551

@JoshuaKGoldberg May remember better, but the opinion I've consistently given is that both `split("")` and spreading are wrong; they are just wrong on different levels, and on most inputs they are both correct, on some inputs the latter is more correct, and on some inputs they are both wrong, and yet in some use cases (namely when you want to get a list of code units for some reason) only the former is correct. There's no reason to prefer one to another and we definitely should not suggest using `split("")` as an alternative without clearly spelling out what each option does. At the end of the day it may actually be best for us to allow string spreading by default and only opt in, and/or only suggesting `Intl.Segmenter`.

**Comment by Josh-Cena** [comment]
(Awaiting response on other team members who may have contributed opinions towards the string spreading case)

**Comment by JoshuaKGoldberg** [comment]
Oh, it's entirely possible that this got lost in all the merges & recreations & threads. Without diving into the old issue & PRs, yes, agreed that we shouldn't suggest using `split("")` as an alternative without the nuance.

My vote, held weakly: let's tweak the rule to just not suggest `split("")`. String spreading in general is super dangerous and weird, and _most_ users should avoid it the way the rule generally suggests (minus the `split("")` suggestion).

I also think as a separate issue it would be good to flesh out the docs in that page. As this issue noted, the whole area of characters and strings is a mess. Prior art: #10227 -> #10731

**Comment by Josh-Cena** [comment]
My thinking is, we are not really doing the user a favor by banning string spreading. In an ideal world when we have `"...".codePoints()` or `"...".codeUnits()`, we may be able to suggest them instead, but at the moment this is the least ambiguous/confusing way to do what's intended most of the time.

**Comment by bradzacher** [comment]
I think:
- add a config option to turn off for the array case (defaults to on)
- remove the recommendation to use `.split`
- include some extra text that helps point out that it's valid (eg "Spreading a string is okay if and only if you are intending to separate the code units").

I think hard removing it would be wrong for the majority of cases -- I think in general spreading a string is a mistake, not intentional, but some codebases it may be common practice.

**Comment by JoshuaKGoldberg** [comment]
Found the discussion around `string`s: https://github.com/typescript-eslint/typescript-eslint/pull/8509#discussion_r1758880247. I align with @Josh-Cena's https://github.com/typescript-eslint/typescript-eslint/pull/8509#discussion_r1826865571:

> Both `split("")` and `...` are only correct under certain assumptions, and practically, most of the time, the former is not much more wrong than the latter. It's fine for unicorn to recommend spreading since it's nominally safer, and if users accept safety at the level of spreading they are free to configure the allow config, but banning it by default exposes such unsafety.

**Comment by JoshuaKGoldberg** [comment]
I sent #10751. Seeking bikeshedding on the report message. 🙂
